### PR TITLE
feat(discv2 results): show/hide tag summary button

### PR DESCRIPTION
**Pull request by doralchan**
**Before:**
Screen Shot 2020-06-06 at 10 26 58 AM

**After:**
Screen Shot 2020-06-06 at 10 28 10 AM
Screen Shot 2020-06-06 at 10 28 17 AM

### CODE REVIEW

File: src/sentry/static/sentry/app/views/eventsV2/table/tableActions.tsx [hunk 1]
```diff
@@ -90,6 +90,15 @@ renderAsyncExportButton.propTypes = {
   isLoading: PropTypes.bool,
 };
 
+function renderSummaryButton() {
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/eventsV2/table/tableActions.tsx:93]: @doralchan if you're looking for feedback; it looks fine design-wise.

Might want to make it more functional though :D

File: src/sentry/static/sentry/app/views/eventsV2/table/tableActions.tsx [hunk 2]
```diff
@@ -90,6 +90,15 @@ renderAsyncExportButton.propTypes = {
...
+  return (
+    <HeaderButton>
+      <IconTag size="xs" />
+      {t('Show Tag Summary')}
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/table/tableActions.tsx:97]: Should this say toggle? Is there a use for showing and then hiding the tag summary?

File: src/sentry/static/sentry/app/views/eventsV2/table/tableActions.tsx [hunk 3]
```diff
@@ -93,98 +102,71 @@ renderAsyncExportButton.propTypes = {
...
-  display: flex;
-  align-items: center;
-
-  background: none;
-  border: none;
-  color: ${p => (p.disabled ? p.theme.gray400 : p.theme.gray600)};
-  cursor: ${p => (p.disabled ? 'default' : 'pointer')};
-  font-size: ${p => p.theme.fontSizeSmall};
+class HeaderActions extends React.Component<Props, State> {
+  state = {showTags: true};
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/table/tableActions.tsx:122]: Could this allow for state to differ between the parent components and this one? For example if the localstorage value is set to 0, the parent component would think that that the sidebar is closed, while this component would have the state as open.

**Review by dashed — APPROVED** [review]
Looks good to me 👍

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/types/index.tsx [hunk 4]
```diff
@@ -997,19 +997,24 @@ export type NewQuery = {
...
   orderby?: string;
+
+  // GlobalSelectionHeader
+  projects: Readonly<number[]>;
+  environment?: Readonly<string[]>;
   range?: string;
   start?: string;
   end?: string;
-  environment?: Readonly<string[]>;
-  tags?: Readonly<string[]>;
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/types/index.tsx:1009]: Is there a reason why `tags?` was deleted here?

Comment (markstory) [reply src/sentry/static/sentry/app/types/index.tsx:1009]: We used to support event views with an explicit list of tags to show in the tag summary sidebar. We removed that feature and replaced it with automatic tag list generation. This is a remnant of the old behavior.

**Pull request merged**